### PR TITLE
add a --app parameter to the default_app

### PR DESCRIPTION
**Pull request by janwiemers**
This pull request implement the idea described in #1877
It worked quite well together with the webdriverio module.

There are currently no tests included because I didn't found the right place.
Maybe someone can point me to the right location so I can attach the tests to the pull request.

cheers Jan

### CODE REVIEW

**Comment by janwiemers** [comment]
Hey there,

can I do something on the failing tests, I didn't changed anything within the build process
But the build didn't worked on my machine (Mac, El Capitan) as well.

**Comment by janwiemers** [comment]
@AlicanC yes you are right the check can be optimized.
I think we should do a check on `argv[i].indexOf(' --app=')` to ensure that the `--app` is a regular parameter.

Whitespaces are no problem, i tested it in my environment.

**Comment by janwiemers** [comment]
@AlicanC you tricked me :) `argv` takes care of splitting the parameter into pieces.
For example `./path/to/my/binary foo --bar --what=ever` ends up in
`argv['foo', '--bar', '--what=ever']` So there is no need to check it again and we can just relay on the current solution just by replacing the `indexOf` against a regular expression to test if the `--app` is the beginning of the string.

**Comment by AlicanC** [comment]
I was actually talking about `electron asd--app=path/to/stuff` and not `electron asd --app=path/to/stuff`.

```bash
> node -e "console.log(process.argv.slice(1));" foo --bar bleh--what=ever
[ 'foo', '--bar', 'bleh--what=ever' ]
```

Arg `bleh--what=ever` doesn't get split to `bleh` and `--what=ever` so if you do a contains check (`indexOf() !== -1`) instead of a starts-with check (`indexOf() === 0`), you will match stuff like `ihavenoidea--app=path/to/app`.

File: atom/browser/default_app/index.html [hunk 1]
```diff
@@ -58,7 +58,7 @@
 <body>
   <script>
     var execPath = require('remote').process.execPath;
-    var command = execPath + ' path-to-your-app';
+    var command = execPath + '--app=path-to-your-app';
```
Comment (zcbenz) [inline atom/browser/default_app/index.html]: I think we should only make `--app` a hack for `webdriverio`, and the `electron path-to-your-app` should still be the preferred way of running apps.

**Comment by janwiemers** [comment]
@AlicanC ok got it, but I think with the regexp it will work while the regular expression checks for the beginning of a string. Or do you have any concerns there?

@zcbenz ok I will change it.

File: atom/browser/default_app/index.html [hunk 2]
```diff
@@ -58,7 +58,7 @@
 <body>
   <script>
     var execPath = require('remote').process.execPath;
-    var command = execPath + ' path-to-your-app';
+    var command = execPath + 'path-to-your-app';
```
Comment (zcbenz) [inline atom/browser/default_app/index.html]: The space should not be removed.

**Comment by janwiemers** [comment]
@zcbenz done :)

**Comment by zcbenz** [comment]
:+1:

**Comment by janwiemers** [comment]
cool thanks :+1:

**Pull request merged**